### ISSUE TITLE
Pex does not correctly resolve dependencies for python3.10

### ISSUE THREAD

**Issue by AndrewHoos** [pantsbuild/pex#1486]
pex seems to consider 3.10 < 3.8

pex-requirements.txt
```
virtualenv==20.8.1
```

Command to reproduce failures
```
python3 -m pex -r pex-requirements.txt -o python --python python3.10 ; PEX_VERBOSE=9 ./python
```

output (truncated):
```
pex: Failed to resolve a requirement: The 'importlib-metadata; python_version < "3.8"' distribution was not found and is required by backports.entry-points-selectable               
pex: Failed to resolve a requirement: The 'pathlib2<3,>=2.3.3; python_version < "3.4" and sys_platform != "win32"' distribution was not found and is required by virtualenv
```

python_version 3.10 should not be less than 3.8 so this requirement should not be needed so it should not fail top resolve

**Comment by AndrewHoos** [comment]
If anyone runs into this it likely because you tried to install pex into python3.10 which is not officially supported, but `pex==2.0.3` installs, but has the bug described. The fix is to uninstall pex from python3.10 and use python3.9 for packaging of python3.10 exes

**Comment by jsirois** [comment]
I'm going to close this as answered since modern Pex resolves for Python 3.10 just fine and runs under Python 3.10 just fine (see the example below that forces running under 3.10):
```console
$ PEX_SCRIPT=virtualenv python3.10 ./python --version
virtualenv 20.8.1 from /home/jsirois/.pex/installed_wheels/7373a3ba8516eea486a88584e9efa940bc50ac5c/virtualenv-20.8.1-py2.py3-none-any.whl/virtualenv/__init__.py
```
Please speak up if this doesn't get you down the line using current Pex to build PEX files that work with Python 3.10.
